contracts/AddrArrayLib.sol

```diff
@@ -0,0 +1,88 @@
+/*
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity >=0.4.25 <0.6.0;
+
+library AddrArrayLib {
+    using AddrArrayLib for Addresses;
+
+    struct Addresses {
+      address[]  _items;
+    }
+
+    /**
+     * @notice push an address to the array
+     * @dev if the address already exists, it will not be added again
+     * @param self Storage array containing address type variables
+     * @param element the element to add in the array
+     */
+    function pushAddress(Addresses storage self, address element) internal {
+      if (!exists(self, element)) {
+        self._items.push(element);
+      }
+    }
+
+    /**
+     * @notice remove an address from the array
+     * @dev it wil find the element and then Swap it with the last element and then delete it
+     * @param self Storage array containing address type variables
+     * @param element the element to remove from the array
+     */
+    function removeAddress(Addresses storage self, address element) internal returns (bool) {
+        for (uint i = 0; i < self.size(); i++) {
+            if (self._items[i] == element) {
+                self._items[i] = self._items[self.size() - 1];
+                self._items.pop();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @notice get the address at a specific index from array
+     * @param self Storage array containing address type variables
+     * @param index the index in the array
+     */
+    function getAddressAtIndex(Addresses storage self, uint256 index) internal view returns (address) {
+        return self._items[index];

```

Comment:
What happens if the index is out of bounds (i.e. >= items.length)?

Reply:
it will revert, I changed the revert msg to "the  index is out of bounds"